autofs-5.0.5 - fix included map read fail handling

From: Ian Kent <raven@themaw.net>

If an included map read fails an error is returned and subsequent
master map entries are not read. We should report the failure but
we shouldn't stop reading the master map.
---

 CHANGELOG             |    4 ++++
 modules/lookup_file.c |    7 -------
 2 files changed, 4 insertions(+), 7 deletions(-)


diff --git a/CHANGELOG b/CHANGELOG
index e734cb3..674a48b 100644
--- CHANGELOG
+++ CHANGELOG
@@ -1,3 +1,7 @@
+??/??/20?? autofs-5.0.6
+-----------------------
+- fix included map read fail handling.
+
 03/09/2009 autofs-5.0.5
 -----------------------
 - fix dumb libxml2 check
diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index a4ca39d..e43ab2f 100644
--- modules/lookup_file.c
+++ modules/lookup_file.c
@@ -438,13 +438,6 @@ int lookup_read_master(struct master *master, time_t age, void *context)
 				     MODPREFIX
 				     "failed to read included master map %s",
 				     master->name);
-				if (!master->recurse) {
-					master->name = save_name;
-					master->depth--;
-					master->recurse = 0;
-					fclose(f);
-					return NSS_STATUS_UNAVAIL;
-				}
 			}
 			master->depth--;
 			master->recurse = 0;
